admin/installer/_images.py

```diff
@@ -67,3 +120,269 @@ def _unserialize_packer_dict(serialized_packer_dict):
         key, value = item.split(":")
         packer_dict[key] = value
     return freeze(packer_dict)
+
+
+class PackerConfigure(PClass):
+    """
+    The attributes necessary to create a custom packer configuration file from
+    the prototype files in ``admin/installer/packer``.
+
+    :ivar build_region: The AWS region to build images in.
+    :ivar publish_regions: The AWS regions to publish the build images to.
+    :ivar template: The prototype configuration to use as a base. One of
+        `docker` or `flocker`.
+    :ivar distribution: The operating system distribution to install.
+        ubuntu-14.04 is the only one implemented so far.
+    :ivar configuration_directory: The directory containing prototype
+        configuration templates.
+    :ivar source_ami: The AMI ID to use as the base image.
+    :ivar working_directory: The directory in which templates will be copied
+        and modified.
+    """
+    build_region = field(type=unicode, mandatory=True)
+    publish_regions = pvector_field(item_type=unicode)
+    template = field(type=unicode, mandatory=True)
+    distribution = field(type=unicode, mandatory=True)
+    configuration_directory = field(type=FilePath, initial=PACKER_TEMPLATE_DIR)
+    source_ami = field(type=unicode, mandatory=True)
+    working_directory = field(type=FilePath, mandatory=True)
+
+
+@sync_performer
+def perform_packer_configure(dispatcher, intent):
+    """
+    Copy the prototype configuration files and provisioning scripts to a
+    temporary location and modify one of the configurations witht the values
+    found in ``intent``.
+    """
+    temporary_configuration_directory = intent.working_directory.child(
+        'packer_configuration'
+    )
+    temporary_configuration_directory.makedirs()
+    intent.configuration_directory.copyTo(temporary_configuration_directory)
+
+    template_name = (
+        u"template_{distribution}_{template}.json".format(
+            distribution=intent.distribution,
+            template=intent.template,
+        )
+    )
+    template_path = temporary_configuration_directory.child(
+        template_name
+    )
+
+    with template_path.open('r') as infile:
+        configuration = json.load(infile)
+
+    # If a region was specified, just build the image there and don't do
+    # any copying.
```

Comment:
I originally had it so that the default was to publish to all regions but if you specified a region it would build in that region without any copying. I removed the comment.